File: JavaScript/260. Single Number III.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var singleNumber = function(nums) {
    var mp = new Map();
    var res = new Array(2);
    for(var i of nums){
        if(mp[i]){
            mp[i]++;
        }
        else{
            mp[i] = 1;
        }
    }
    var idx = 0;
    for(var i of nums){
        if(mp[i]==1){
            res[idx] = i;
            idx++;
        }
    }
    return res;
};

File: Java/260. Single Number III.java
class Solution {
    public int[] singleNumber(int[] nums) {
        HashMap<Integer, Integer> mp = new HashMap();
        int[] res = new int[2];
        for(int i: nums){
            if(mp.containsKey(i)){
                mp.put(i, mp.get(i)+1);
            }
            else{
                mp.put(i,1);
            }
        }
        int idx = 0;
        for(int i: nums){
            if(mp.get(i)==1){
                res[idx] = i;
                idx++;
            }
        }
        return res;
    }
}
